pkg/controller/registry/resolver/solver/tracer.go

```diff
@@ -0,0 +1,33 @@
+package solver
+
+import (
+	"fmt"
+)
+
+type SearchPosition interface {
+	Installables() []Installable
+	Conflicts() []AppliedConstraint
+}
+
+type Tracer interface {
+	Trace(p SearchPosition)
+}
+
+type DefaultTracer struct{}
+
+func (DefaultTracer) Trace(_ SearchPosition) {
+}
+
+type PrintfTracer struct{}
+
+func (PrintfTracer) Trace(p SearchPosition) {
```

Comment:
`Tracer` could just embed `io.Writer` and then it would be easy to write tests for the Tracer / validate its output

non blocking, but it would be nice to see the trace output for unsatisfiable tests

Reply:
> non blocking, but it would be nice to see the trace output for unsatisfiable tests

Done, good idea.